File: genetic-analyzer/bin/de_analysis.py
#!/usr/bin/env python

# Copyright (C) 2017 by
# Thomas E. Gorochowski <tom@chofski.co.uk>, Voigt Lab, MIT
# All rights reserved
# Released under MIT license (see LICENSE.txt)

# Supporting modules
import argparse
import genetic_analyzer as ga

def main():
	# Parse the command line inputs
	parser = argparse.ArgumentParser(description="de_analysis")
	parser.add_argument("-settings",  dest="settings",  required=True,  help="settings.txt", metavar="string")
	parser.add_argument("-group1",  dest="group1",  required=True,  help="1,2,6,7", metavar="string")
	parser.add_argument("-group2",  dest="group2",  required=True,  help="3,4,8,9", metavar="string")
	parser.add_argument("-output_prefix",  dest="output_prefix",  required=True,  help="WT_vs_Sample", metavar="string")
	parser.add_argument("-bin_path",  dest="bin_path",  required=True,  help="/usr/bin/", metavar="string")
	args = parser.parse_args()
	# Run the command
	settings = ga.load_settings(args.settings)
	cur_group1 = args.group1
	cur_group2 = args.group2
	cur_output_prefix = args.output_prefix
	cur_bin_path = args.bin_path
	# Run the DE analysis
	ga.de_analysis(settings, cur_group1, cur_group2, cur_output_prefix, bin_path=cur_bin_path)

if __name__ == "__main__":
	main()

File: genetic-analyzer/bin/de_analysis.r
#!/usr/bin/env Rscript

# Copyright (C) 2017 by
# Thomas E. Gorochowski <tom@chofski.co.uk>, Voigt Lab, MIT
# All rights reserved
# Released under MIT license (see LICENSE.txt)

# Arguments:
# 1. count_matrix_filename
# 2. group1, e.g., 1,2,5,6
# 3. group2, e.g., 3,4,7,8
# 4. library_size_matrix
# 5. output_file_prefix
args <- commandArgs(TRUE)
arg_count_matrix <- args[1]
arg_group1 <- as.integer(unlist(strsplit(args[2], ",")))
arg_group2 <- as.integer(unlist(strsplit(args[3], ",")))
arg_library_size_matrix <- args[4]
arg_output_file_prefix <- args[5]

# We use edgeR for DE analysis
library(edgeR)

# Load the data file
count_matrix <- read.table(arg_count_matrix, header=T, row.names=1, com='')

# Extract those columns we are interested in
col_ordering <- c(arg_group1, arg_group2)
count_matrix <- count_matrix[,col_ordering]
conditions <- factor(c(rep("group_1", length(arg_group1)), rep("group_2", length(arg_group2))))

# Load the actual mapped reads
library_size_matrix <- read.table(arg_library_size_matrix, header=T, row.names=1, com='')

# Get the library sizes (total counts in annotated genes)
lib_sizes <- as.vector(library_size_matrix$total_mapped_reads)[col_ordering]

# Calculate normalisation factors using TMM
expr <- DGEList(counts=count_matrix, group=conditions, lib.size=lib_sizes)
expr <- calcNormFactors(expr)
expr <- estimateCommonDisp(expr)
expr <- estimateTagwiseDisp(expr)
de_data <- exactTest(expr)
de_results <- topTags(de_data, n=length(count_matrix[,1]))

# write the output to a text file
write.table(as.matrix(de_results$table), col.names=NA, quote=FALSE, file=paste0(arg_output_file_prefix, ".de.analysis.txt"), sep="\t")
